Framework/LiveData/inc/MantidLiveData/ISIS/ISISHistoDataListener.h

```diff
@@ -131,6 +129,10 @@ class ISISHistoDataListener : public API::LiveListener {
   /// Time regime to load
   int m_timeRegime;
 
+  /// Buffer workspace to store instrument data (or not only instrument in the
+  /// future), prevents loading for every chank of data
+  API::MatrixWorkspace_sptr bufferWorkspace;
```

Comment:
This will work, however, since you only want to cache the instrument, storing the instrument const shared pointer, fetched via `getInstrument()` and set via `setInstrument()` https://github.com/mantidproject/mantid/blob/master/Framework/API/inc/MantidAPI/ExperimentInfo.h#L66-L68 could be an alternative approach to your methodology here.